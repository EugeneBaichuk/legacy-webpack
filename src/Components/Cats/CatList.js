import React from 'react';
import { ListGroup, ListGroupItem } from 'reactstrap';

const CatList = ({cats}) => (
  <ListGroup>
    {cats.map((cat, i) => <ListGroupItem key={i}>{cat}</ListGroupItem>)}
  </ListGroup>
)

export default CatList;
