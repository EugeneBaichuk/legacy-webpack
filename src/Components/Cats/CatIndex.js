import React from 'react'
import CatList from './CatList'

const CatIndex = ({header, list}) => (
  <section>
    <h1>{header}</h1>
    <CatList cats={list}/>
  </section>
)

export default CatIndex
