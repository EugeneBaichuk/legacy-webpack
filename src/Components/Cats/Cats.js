import React from 'react';
import Layout from '../Common/Layout';
import CatIndex from './CatIndex'
import { catsHeader, catsTaskList } from '../../Config/cats.constants';
import { breedsList, catsText } from '../../Config/cats.constants'

export const Cats = () => (
  <Layout 
    header={catsHeader} 
    list={catsTaskList}>
    <CatIndex list={breedsList} header={catsText} /> 
  </Layout>
)

