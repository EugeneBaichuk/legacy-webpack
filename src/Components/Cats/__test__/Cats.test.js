import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Cats } from '../Cats';

describe('Cats component', () => {
  afterEach(cleanup);

  test('renders the header correctly', () => {
    const { asFragment } = render(<Cats />);
    expect(asFragment(<Cats />)).toMatchSnapshot();
  });
});