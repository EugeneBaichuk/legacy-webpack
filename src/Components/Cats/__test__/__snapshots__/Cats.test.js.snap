// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cats component renders the header correctly 1`] = `
<DocumentFragment>
  <div
    class="layout container"
  >
    <h1
      class="mb-4 mt-4"
    >
      To complete this challenge, complete and fix the existing code to have it do the following:
    </h1>
    <ul
      class="list-group"
    >
      <li
        class="list-group-item"
      >
        Render the list of cat breeds to the screen
      </li>
      <li
        class="list-group-item"
      >
        Use props correctly
      </li>
      <li
        class="list-group-item"
      >
        Use .map() correctly
      </li>
      <li
        class="list-group-item"
      >
        Ensure there are no errors
      </li>
    </ul>
    <hr
      class="my-4"
    />
    <section>
      <h1>
        Raining Cats!
      </h1>
      <ul
        class="list-group"
      >
        <li
          class="list-group-item"
        >
          persian
        </li>
        <li
          class="list-group-item"
        >
          siamese
        </li>
        <li
          class="list-group-item"
        >
          maine coon
        </li>
        <li
          class="list-group-item"
        >
          ragdoll
        </li>
        <li
          class="list-group-item"
        >
          scottish fold
        </li>
        <li
          class="list-group-item"
        >
          sphynx
        </li>
        <li
          class="list-group-item"
        >
          british shorthair
        </li>
        <li
          class="list-group-item"
        >
          bengal
        </li>
        <li
          class="list-group-item"
        >
          american shorthair
        </li>
      </ul>
    </section>
  </div>
</DocumentFragment>
`;
